iconservice/icon_service_engine.py

```diff
@@ -317,17 +330,60 @@ def invoke(self,
                 block_result.append(tx_result)
                 context.block_batch.update(context.tx_batch)
                 context.tx_batch.clear()
+                precommit_flag = self._get_precommit_step_flag(precommit_flag, tx_result)
+
+                if precommit_flag & PrecommitFlag.STEP_ALL_CHANGED != PrecommitFlag.NONE:
+                    self._update_step_properties(context, precommit_flag)
 
         # Save precommit data
         # It will be written to levelDB on commit
         precommit_data = PrecommitData(
-            context.block_batch, block_result, context.new_icon_score_mapper)
+            context.block_batch, block_result, context.new_icon_score_mapper, precommit_flag)
         self._precommit_data_manager.push(precommit_data)
 
         self._context_factory.destroy(context)
 
         return block_result, precommit_data.state_root_hash
 
+    @staticmethod
+    def _get_precommit_step_flag(precommit_flag, tx_result):
+        if tx_result.score_address == GOVERNANCE_SCORE_ADDRESS:
+            # Governance is updated last tx, Updates STEP properties
+            precommit_flag = PrecommitFlag.STEP_ALL_CHANGED
+        else:
+            for event_log in tx_result.event_logs:
+                if event_log.score_address == GOVERNANCE_SCORE_ADDRESS:
+                    if event_log.indexed[0] == 'StepPriceChanged(int)':
+                        precommit_flag |= PrecommitFlag.STEP_PRICE_CHANGED
+
+                    if event_log.indexed[0] == 'StepCostChanged(str,int)':
+                        precommit_flag |= PrecommitFlag.STEP_COST_CHANGED
+
+                    if event_log.indexed[0] == 'MaxStepLimitChanged(str,int)':
+                        precommit_flag |= PrecommitFlag.STEP_MAX_LIMIT_CHANGED
+
+        return precommit_flag
+
+    def _update_step_properties(self, context, precommit_flag):
+
```

Comment:
first should call self._push_context(context) before that execute this logic, if you want access properties in SCORE or call call something

Reply:
Added it